lib/arboreal/instance_methods.rb

```diff
@@ -83,9 +83,7 @@ def sibling_conditions
 
     def populate_materialized_path
       if parent_id_changed? || materialized_path.nil?
-        model_base_class.send(:with_exclusive_scope) do
-          self.materialized_path = parent ? parent.path_string : "-"
-        end
+        self.materialized_path = (parent_id && parent) ? parent.path_string : "-"
```

Comment:
Is it possible for `parent` to exist but not have `parent_id`, such as during object building with new records?